File: Chapter08/airline-fare/src/main/java/springfive/airline/airlinefare/infra/redis/RedisConfiguration.java
package springfive.airline.airlinefare.infra.redis;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.redis.connection.ReactiveRedisConnectionFactory;
import org.springframework.data.redis.core.ReactiveRedisTemplate;
import org.springframework.data.redis.serializer.RedisSerializationContext;

@Configuration
public class RedisConfiguration {

  @Bean
  public ReactiveRedisTemplate<String,String> reactiveRedisTemplate(ReactiveRedisConnectionFactory reactiveRedisConnectionFactory){
    return new ReactiveRedisTemplate<>(reactiveRedisConnectionFactory, RedisSerializationContext.string());
  }

  @Bean
  public ObjectMapper mapper(){
    return new ObjectMapper();
  }

}

File: Chapter04/tracked-hashtag-service/src/main/kotlin/springfive/twittertracked/infra/redis/RedisConfiguration.kt
package springfive.twittertracked.infra.redis

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.redis.connection.ReactiveRedisConnectionFactory
import org.springframework.data.redis.core.ReactiveRedisTemplate
import org.springframework.data.redis.serializer.RedisSerializationContext

/**
 * @author claudioed on 06/12/17.
 */
@Configuration
open class RedisConfiguration {

    @Bean
    open fun reactiveRedisTemplate(connectionFactory: ReactiveRedisConnectionFactory): ReactiveRedisTemplate<String, String> {
        return ReactiveRedisTemplate(connectionFactory, RedisSerializationContext.string())
    }

}